src/bin/rcc.rs

```diff
@@ -125,17 +125,22 @@ fn real_main(
         if let Some(main) = rccjit.get_compiled_function("main") {
             let args = std::env::args();
             let argc = args.len() as i32;
-            let vec_args = args
-                .map(|string| std::ffi::CString::new(string).unwrap())
-                .collect::<Vec<std::ffi::CString>>();
-            let pointer = vec_args
-                .iter()
-                .map(|cstr| cstr.as_ptr() as *const u8)
-                .collect::<Vec<*const u8>>()
-                .as_ptr() as *const *const u8;
-            let main: unsafe extern "C" fn(i32, *const *const u8) -> i32 =
-                unsafe { std::mem::transmute(main) };
-            let _ = unsafe { main(argc, pointer) };
+            let exit_code = {
+                // we use block there so we can be 100% sure that memory allocated for CString is freed.
+                let vec_args = args
+                    .map(|string| std::ffi::CString::new(string).unwrap())
+                    .collect::<Vec<std::ffi::CString>>();
+                // CString should be alive if we want to pass it's pointer to another function properly, otherwise this may lead to memory leak or UB.
+                let pointer = vec_args
+                    .iter()
+                    .map(|cstr| cstr.as_ptr() as *const u8)
+                    .collect::<Vec<*const u8>>()
+                    .as_ptr() as *const *const u8;
+                let main: unsafe extern "C" fn(i32, *const *const u8) -> i32 =
+                    unsafe { std::mem::transmute(main) }; // this transmute is safe: this function is finalized(`rccjit.finalize()`) and **guaranteed** to be non-null
+                unsafe { main(argc, pointer) }
+            };
+            std::process::exit(exit_code);
```

Comment:
Now that you've mentioned releasing resources, this doesn't release `Files` either ... But we can fix that in a follow-up PR, it should be done by the OS anyway.